upgrade_policy/quorum_upgrade/scripts/ts/src/transactions.ts

```diff
@@ -67,6 +69,38 @@ const proposeUpgrade = (txb: TransactionBlock, quorumUpgradeCapId: string, packa
 
 }
 
+/// Calls `quorum_upgrade_policy::propose_upgrade_v2`, `quorum_upgrade_policy::add_all_metadata` (optional), `quorum_upgrade_policy::share_upgrade_object`,  
+/// Calling this will publish the `ProposedUpgrade` shared object.
+const proposeUpgradeV2 = (txb: TransactionBlock, quorumUpgradeCapId: string, packagePath: string, metadata = new Map<string, Uint8Array>) => {
```

Comment:
You could default to null and make the param optional (`?`) It's TS :)

Reply:
Also instead of a Map, we could use a plain object format (and get keys, values).